### PR TITLE
Add clear error message when Reactive REST Client used outside of Quarkus

### PR DESCRIPTION
**Pull request by geoand**
Relates to: #35038

### CODE REVIEW

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/RedirectHandler.java [hunk 1]
```diff
@@ -14,7 +15,42 @@ public interface RedirectHandler {
...
+                        return null;
+                    }
+                };
+            }
+        }
+
+        interface Context {
+            Response response();
+        }
+    }
```
Comment (Ladicek) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/RedirectHandler.java:49]: This doesn't seem relevant to the present PR? :-)

**Review by Ladicek — APPROVED** [review]

**Review by Sgitario — APPROVED** [review]
LGTM
